File: 1218-longest-arithmetic-subsequence-of-given-difference/1218-longest-arithmetic-subsequence-of-given-difference.cpp
class Solution {
public:
    int longestSubsequence(vector<int>& arr, int difference) {
        return solve(arr, difference);
    }
private:
    int solve(vector<int>& arr, int diff) {
        unordered_map<int, int> hashMap;
        int ans = 1;
        
        for(int i = 0; i < arr.size(); ++i) {
            hashMap[arr[i]] = hashMap[arr[i] - diff] + 1;
            ans = max(ans, hashMap[arr[i]]);
        }
        return ans;
    }
};

File: 1218-longest-arithmetic-subsequence-of-given-difference/1218-longest-arithmetic-subsequence-of-given-difference.py
class Solution:
    def longestSubsequence(self, arr: List[int], difference: int) -> int:
        hashMap = dict()
        ans = 1
        
        for n in arr:
            hashMap[n] = hashMap.get(n - difference, 0) + 1
            ans = max(ans, hashMap[n])
        return ans
            
        